domain/schema_validator.go

```diff
@@ -173,6 +173,12 @@ func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedTableIDs [
 
 	// Schema changed, result decided by whether related tables change.
 	if schemaVer < s.latestSchemaVer {
+		// The DDL relatedTableIDs is empty.
+		if len(relatedTableIDs) == 0 {
+			log.Infof("[domain-ddl] the related table ID is empty, checking %d", schemaVer)
+			return ResultFail
```

Comment:
Why return fail when the DDL relatedTableIDs is empry?

Reply:
The DDL relatedTableIDs is always empty. If `schemaVer < s.latestSchemaVer` is true when executing DDL，we return `ResultFail`.
The DDL statements get relatedTableIDs is complicated, so I didn't do it.